homeassistant/components/shelly/utils.py

```diff
@@ -346,3 +346,30 @@ def get_rpc_input_triggers(device: RpcDevice) -> list[tuple[str, str]]:
             triggers.append((trigger_type, subtype))
 
     return triggers
+
+
+async def async_device_update_info(
```

Comment:
I think this does not need to be async and we can remove `await` in calls
```suggestion
def async_device_update_info(
```

Reply:
There are 3 async all inside async_device_update_info

Reply:
Just make it an async callback. Decorate it with `@callback` to make it clear.

Reply:
Done